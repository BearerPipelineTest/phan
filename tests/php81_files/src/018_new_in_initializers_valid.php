<?php

namespace NS18;

use ArrayObject;
use Attribute;

#[Attribute]
class ExampleAttribute {
    public function __construct(
        public ArrayObject $values = new ArrayObject(['first'])
    ) {}
}

const Values = new ArrayObject();
static $values = new ArrayObject();
static $e = new ExampleAttribute($values);

#[ExampleAttribute(new ArrayObject(array: [1]))]
class Other {
}

