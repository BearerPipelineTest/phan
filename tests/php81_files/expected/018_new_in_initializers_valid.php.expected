%s:10 PhanTypeMismatchDefault Default value for \ArrayObject $values can't be array|bool|float|int|null|resource|string
%s:11 PhanInvalidConstantExpression Constant expression contains invalid operations
%s:11 PhanWriteOnlyPublicProperty Possibly zero read references to public property \NS18\ExampleAttribute->values
%s:15 PhanInvalidConstantExpression Constant expression contains invalid operations
%s:16 PhanInvalidConstantExpression Constant expression contains invalid operations
%s:17 PhanInvalidConstantExpression Constant expression contains invalid operations
%s:19 PhanInvalidConstantExpression Constant expression contains invalid operations
%s:20 PhanUnreferencedClass Possibly zero references to class \NS18\Other
